Client/CloudAppClient.php

```diff
@@ -36,6 +36,14 @@ public function upload($archive)
     /**
      * {@inheritdoc}
      */
+    public function download()
+    {
+        throw new \LogicException(sprintf('Download method for %s is not implemented.', __CLASS__));
```

Comment:
Hm. I'm not too happy with this. It violates the liskov substitution principle and the interface segregation principle. Maybe add a new interface with the download method? Adding a new interface would not break BC.